src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -981,6 +987,8 @@ public synchronized FeedbackOutputStream getOutput(
               throws IOException {
             // caller will be the exclusive owner of this write stream. all other requests
             // will block until it is returned via a close.
+            // Map identifier to digest function
+            digest_to_function_map.put(key.getDigest().getHash(), key.getDigestFunction());
```

Comment:
This seems like a work around to store the hashing algorithm for digests. It also introduces extra overhead to maintain a cache.